IPython/core/oinspect.py

```diff
@@ -633,26 +671,37 @@ def pinfo(self, obj, oname='', formatter=None, info=None, detail_level=0):
 
         - oname: name of the variable pointing to the object.
 
-        - formatter: special formatter for docstrings (see pdoc)
+        - formatter: callable (optional)
+              A special formatter for docstrings.
+
+              The formatter is a callable that takes a string as an input
+              and returns either a formatted string or a mime type bundle
+              in the form of a dictionnary.
+
+              Although the support of custom formatter returning a string
+              instead of a mime type bundle is deprecated.
 
         - info: a structure with some information fields which may have been
           precomputed already.
 
         - detail_level: if set to 1, more information is given.
         """
-        text = self._format_info(obj, oname, formatter, info, detail_level)
-        if text:
-            page.page(text)
-    
+        info = self._get_info(obj, oname, formatter, info, detail_level)
+        if info:
+            page.page(info)
```

Comment:
So `pinfo` here is API and my guess is it should still `page` the text  instead of the mime bundle, as pagers are expecting text (in terminal for example). 

Should be add a method `rich_page`, that take a mimebundle, and call it if present ?  But I doubt we can just change what we can pass to page or all hell will break loose, and as it has already frozen with bash on windows I dont' want to have to deal with angry cold deamons.